sycl/include/sycl/detail/kernel_desc.hpp

```diff
@@ -151,6 +152,107 @@ template <class KernelNameType> struct KernelInfo {
 };
 #endif //__SYCL_UNNAMED_LAMBDA__
 
+template <typename KNT> struct KernelIdentity {
+  using type = KNT;
+};
+
+template <typename KernelNameType> constexpr unsigned getKernelNumParams() {
+#if __has_builtin(__builtin_sycl_kernel_param_count) &&                        \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  __builtin_sycl_kernel_param_count(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getNumParams();
+#endif
+}
+
+template <typename KernelNameType>
+std::vector<kernel_param_desc_t> getKernelParamDescs() {
+  std::vector<kernel_param_desc_t> Result;
+  int NumParams = getKernelNumParams<KernelNameType>();
+  Result.reserve(NumParams);
+  for (int I = 0; I < NumParams; ++I) {
+#if __has_builtin(__builtin_sycl_kernel_name) &&                               \
+    __has_builtin(__builtin_sycl_kernel_param_access_target) &&                \
+    __has_builtin(__builtin_sycl_kernel_param_size) &&                         \
+    __has_builtin(__builtin_sycl_kernel_param_offset) &&                       \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+    kernel_param_desc_t ParamDesc;
+    ParamDesc.kind =
+        __builtin_sycl_kernel_param_kind(KernelIdentity<KernelNameType>(), I);
+    ParamDesc.info = Result.kind == kind_accessor
```

Comment:
You need to index the vector and add the scope `kernel_param_kind_t::` for` kind_accessor`